framework/include/utils/MooseADWrapper.h

```diff
@@ -25,16 +25,24 @@ struct MooseADWrapperStruct<Real, true>
   typedef ADReal type;
 };
 
+template <>
+struct MooseADWrapperStruct<Point, true>
+{
+  typedef ADPoint type;
+};
+
+// W<T> -> W<ADT>, e.g. std::vector<Real> -> std::vector<ADReal>
 template <template <typename> class W, typename T, bool is_ad>
 struct MooseADWrapperStruct<W<T>, is_ad>
 {
   typedef W<typename MooseADWrapperStruct<T, is_ad>::type> type;
 };
 
-template <typename T, bool is_ad>
-struct MooseADWrapperStruct<std::vector<T>, is_ad>
```

Comment:
I guess `std::vector` isn't actually currently being used anywhere? Because this new struct surely won't match

Reply:
Doesn't https://github.com/idaholab/moose/pull/20200/files#diff-97b9b30d08fb2ef8e6d28995cd9a02639e2f0f1aa94cb0d8e77380d3d1092284R34-R39 already match `std::vector<T>`?!

Reply:
`DynamicStdArrayWrapper::DynamicStdArrayWrapper(std::size_t n)` does not exist. And The MooseADWrapper template that works for `std::array` doesn't apply (because of the `NType`  vs. `std::size_t` template argument). All that could be worked around. But color me concerned about the `MooseADWrapper<DynamicStdArrayWrapper>::type` issue.